iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java

```diff
@@ -78,33 +70,14 @@ public String[] getLocations() {
     // getLocations() won't be accurate when called on worker nodes and will always return "*"
     if (locations == null && conf != null) {
       boolean localityPreferred = conf.getBoolean(InputFormatConfig.LOCALITY, false);
-      locations = localityPreferred ? blockLocations(taskGroup, conf) : ANYWHERE;
+      locations = localityPreferred ? Util.blockLocations(taskGroup, conf) : ANYWHERE;
     } else {
       locations = ANYWHERE;
     }
 
     return locations;
   }
 
-  // We should move to Util.blockLocations once the following PR is merged and shipped
-  // https://github.com/apache/iceberg/pull/11053
-  private static String[] blockLocations(ScanTaskGroup<FileScanTask> task, Configuration conf) {
-    final Set<String> locationSets = Sets.newHashSet();
-    task.tasks().forEach(fileScanTask -> {
-      final Path path = new Path(fileScanTask.file().path().toString());
-      try {
-        final FileSystem fs = path.getFileSystem(conf);
-        for (BlockLocation location : fs.getFileBlockLocations(path, fileScanTask.start(), fileScanTask.length())) {
-          locationSets.addAll(Arrays.asList(location.getHosts()));
-        }
-      } catch (IOException e) {
-        LOG.warn("Failed to get block locations for path {}", path, e);
-      }
-    });
-
-    return locationSets.toArray(new String[0]);
-  }
-
```

Comment:
Just for your information, [this is the replacement as of 1.9.2](https://github.com/apache/iceberg/blob/apache-iceberg-1.9.2/core/src/main/java/org/apache/iceberg/hadoop/Util.java#L61-L76).

Reply:
@okumin shouldn't we upgrade the iceberg version? We are on 1.9.1 ATM
https://github.com/apache/hive/blob/master/iceberg/pom.xml#L29

Reply:
Regarding this context, no, we shouldn't. My patch is included in 1.8.0 or newer versions.